src/config.py

```diff
@@ -48,8 +48,24 @@ def is_build_enabled():
     return parse_dlc_developer_configs("build", "do_build")
 
 
-def is_autopatch_build_enabled():
-    return parse_dlc_developer_configs("build", "autopatch_build") or os.getenv("AUTOPATCH")
+def is_autopatch_build_enabled_in_image_buildspec(buildspec_path=None):
+    from buildspec import Buildspec
```

Comment:
As of now, our `src` (build logic) and `test` (test logic) cross-leverage functionalities. Consequently, we run into issues like circular imports and imports that might not be needed by one logic. To prevent this scenario, we use lazy imports.

In general, it is a good practice to have lazy imports as it increases the startup time and reduces memory consumption. ([reference](https://www.tutorialspoint.com/lazy-import-in-python#:~:text=Benefits%20of%20Lazy%20Import&text=Faster%20startup%20time%20%E2%88%92%20Your%20Python,just%20importing%20modules%20when%20necessary.))